dashboard/src/test/java/com/google/cloud/tools/opensource/dashboard/DashboardTest.java

```diff
@@ -102,16 +98,25 @@ public static void cleanUp() {
   }
 
   @Test
-  public void testCss()
-      throws IOException, TemplateException, ParsingException, ArtifactDescriptorException {
+  public void testCss() {
     Path dashboardCss = outputDirectory.resolve("dashboard.css");
     Assert.assertTrue(Files.exists(dashboardCss));
     Assert.assertTrue(Files.isRegularFile(dashboardCss));
   }
 
+  private void assertDocument(String fileName, Consumer<Document> assertion)
+      throws IOException, ParsingException {
+    Path html = outputDirectory.resolve(fileName);
+    Assert.assertTrue(Files.isReadable(html));
+
+    try (InputStream source = Files.newInputStream(html)) {
+      Document document = builder.build(source);
+      assertion.accept(document);
+    }
+  }
+
   @Test
-  public void testDashboard()
-      throws IOException, TemplateException, ParsingException, ArtifactDescriptorException {
+  public void testDashboard() throws IOException, ParsingException, ArtifactDescriptorException {
```

Comment:
The reason why `testDashboard` is not using `assertDocument` is because the assertion part has function calls with checked exception and not suitable for a lambda expression for 2nd argument of `assertDocument`

Reply:
I don't think you need the lambda or this method  if you set up the document once in setUp